processor/tailsamplingprocessor/processor.go

```diff
@@ -381,6 +391,9 @@ func (tsp *tailSamplingSpanProcessor) processTraces(resourceSpans ptrace.Resourc
 
 		// The only thing we really care about here is the final decision.
 		actualData.Lock()
+		if tsp.sampledIDCache.Get(id) {
```

Comment:
I think we can short-circuit it: if we have a decision, we don't need to build a new policy map, we can just send the trace to the next consumer. Perhaps we need a new metric, to record how many spans were spared from the waiting time?

Reply:
Good idea, I've moved it to the very first section of the loop